extensions/ql-vscode/src/qltest-discovery.ts

```diff
@@ -177,6 +174,15 @@ export class QLTestDiscovery extends Discovery<QLTestDiscoveryResults> {
     this._onDidChangeTests.fire();
   }
 
+  /**
+   * Only include qlpacks suffixed with '-test' that are contained
+   * within the provided workspace folder.
+   */
+  private isRelevantQlPack(qlPack: QLPack): boolean {
+    return qlPack.name.endsWith('-tests')
+      && qlPack.uri.fsPath.startsWith(this.workspaceFolder.uri.fsPath);
```

Comment:
Just making sure: Are the URI paths normalised the same way / always on the same filesystem? Just want to be sure there's no weirdness with the pack URI and the folder URI not matching up, e.g. on Windows.

Reply:
The Uri `fsPath` property will normalize the path according to the OS path separator. This should not be a problem. See https://code.visualstudio.com/api/references/vscode-api#Uri

Reply:
```suggestion
      && (workspace.getWorkspaceFolder(qlpack.uri) === this.workspaceFolder);
```
Even easier and cleaner.

Reply:
Unfortunately, that won't work because there's no guarantee that two workspace folders are the same object. Though, something like this should work and is almost as clean.

    

```suggestion
      && workspace.getWorkspaceFolder(qlPack.uri)?.index === this.workspaceFolder.index;
```

Reply:
It's a better solution, but makes the tests slightly uglier.

Reply:
I think comparing `.index` should be correct, but I thought it was the case that vscode guaranteed that a workspace folder was uniquely characterized by its url --- so in that case wouldn't it also suffice to compare equality of
`workspace.getWorkspaceFolder(qlpack.uri).uri.toString()`
and
`this.workspaceFolder.uri.toString()`
?

If that makes testing easier since you don't need to control index order, I suggest that, otherwise, `.index` is fine.